tests/main/dbus-activation-session/task.yaml

```diff
@@ -0,0 +1,49 @@
+summary: D-Bus session services support activation via systemd user session
+
+systems:
+    # Ubuntu Core does not contain the D-Bus user session systemd units
+    - -ubuntu-core-*
+    # Ubuntu 14.04 does not provide user@.service
+    - -ubuntu-14.04-*
+    # CentOS/Amazon Linux give error "Unit user@0.service not loaded."
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    snap set system experimental.dbus-activation=true
+    #shellcheck source=tests/lib/pkgdb.sh
+    . "$TESTSLIB"/pkgdb.sh
+    distro_install_package dbus-user-session
+    session-tool -u test --prepare
+
+restore: |
+    session-tool -u test --restore
+    snap unset system experimental.dbus-activation
+    snap unset system experimental.user-daemons
+
+execute: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    echo "Install a snap containing an activatable D-Bus session service"
+    snap install --edge test-snapd-dbus-service
```

Comment:
Does the snap need to come from the store for some assertions?

Reply:
The snap is a Python daemon making use of packages not included in the base snap, so couldn't simply be built with `snap pack`.  So it is instead built with Snapcraft and the result published to the store.